implementation/src/main/java/io/smallrye/config/AbstractLocationConfigSourceLoader.java

```diff
@@ -264,7 +264,13 @@ private boolean validExtension(final String resourceName) {
 
     private static URI addProfileName(final URI uri, final String profile, final String fileExtension) {
         if ("jar".equals(uri.getScheme())) {
-            return URI.create("jar:" + addProfileName(URI.create(uri.getRawSchemeSpecificPart()), profile, fileExtension));
```

Comment:
It is `+`, not concatenation, which incurs the cost; using `String.concat()` is simpler and cleaner here. But, even better would be to pass the option to `javac` to disable indy string concatenation and leave the code as-is IMO. That way we won't miss any.

Reply:
The option is `-XDstringConcat=inline`.

Reply:
@dmlloyd Thanks, I was looking for that `javac` flag! Were did you find that?

Reply:
It is used by openjdk when building the `java.base` module.

Reply:
Cool, thanks!

So passing configuring Maven on `smallrye-config` to use the flag seems like a good option, but the real question is do we want to do that for Quarkus itself.... My feeling is that we should only do that for the core runtime module, WDYT?

Reply:
Yeah we should be careful where we apply it. And I think there will be some cases where `String.concat` will be faster too.

Reply:
I think applying it to this moduke and `quarkus-core` is safe, but I'm fine using `.concat` instead

Reply:
Honestly I don't know why javac would ever produce anything other than `.concat` for simple `a + b` (just single concat) situations.

Reply:
Yeah, sometimes the mundane approach makes a lot more sense than the fancy / complicated one...